### PR TITLE
#900 Add a structure to be able to know if a z is connected or not in…

### PR DESCRIPTION
**Pull request by rosiereflo**
…side a model. Small optims on network models.

### CODE REVIEW

**Comment by rosiereflo** [comment]
This provides a gain of ~5s on a large t0.

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelLoad.cpp [hunk 1]
```diff
@@ -275,13 +279,17 @@ ModelLoad::evalJt(SparseMatrix& jt, const double& cj, const int& rowOffset) {
...
       } else if ((zq > 0. && zq < zQMax_) || (zq <= 0. && zQdiff > 0.) || (zq >= zQMax_ && zQdiff < 0.)) {
-        double termZq = -pow(U, beta_) * kq_;
-        double termUr = betaLong_ * ur * pow(U, betaLong_ - 2.) / pow(u0_, betaLong_) - zq * beta_ * ur * pow(U, beta_ - 2.) * kq_;
-        double termUi = betaLong_ * ui * pow(U, betaLong_ - 2.) / pow(u0_, betaLong_) - zq * beta_ * ui * pow(U, beta_ - 2.) * kq_;
+        double termZq = -powUBeta * kq_;
+        double powUAlphaLong = pow(U, betaLong_ - 2.);
+        double powU0AlphaLong = pow(u0_, betaLong_);
+        double powUAlphaMinus2 = pow(U, beta_ - 2.);
+        double termUr = betaLong_ * ur * powUAlphaLong / powU0AlphaLong - zq * beta_ * ur * powUAlphaMinus2 * kq_;
+        double termUi = betaLong_ * ui * powUAlphaLong / powU0AlphaLong - zq * beta_ * ui * powUAlphaMinus2 * kq_;
```
Comment (adriguir) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelLoad.cpp:292]: Adapt names

File: dynawo/sources/Modeler/Common/DYNModelMulti.cpp [hunk 2]
```diff
@@ -196,6 +200,10 @@ ModelMulti::initBuffers() {
   yLocal_ = new double[sizeY_]();
   ypLocal_ = new double[sizeY_]();
   zLocal_ = new double[sizeZ_]();
+  zConnectedLocal_ = new bool[sizeZ_];
+  for (int i = 0; i < sizeZ_; ++i) {
+    zConnectedLocal_[i] = false;
+  }
```
[the remark this answers is not in the document]
Comment (rosiereflo) [reply dynawo/sources/Modeler/Common/DYNModelMulti.cpp:206]: Let me see if there is a more elegant way to initialize this in the stl.
It is really needed as default initialization is dependent from the compiler, the version of the compiler, the OS ... Most of them must initialize bool to false, but some others might just leave the values in memory.

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelLoad.cpp (hunk 1 above)
Comment (rosiereflo) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNModelLoad.cpp]: Solved

File: dynawo/sources/Modeler/Common/DYNModelMulti.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (rosiereflo) [reply dynawo/sources/Modeler/Common/DYNModelMulti.cpp]: Solved

**Pull request merged**